src/mono/wasm/wasm.proj

```diff
@@ -70,11 +73,90 @@
+      <_EmccFlags Include="-s ALIASING_FUNCTION_POINTERS=0" />
+      <_EmccFlags Include="-s USE_ZLIB=1" />
+      <_EmccFlags Include="-DENABLE_NETCORE=1" Condition="'$(Configuration)' == 'Release'" />
+    </ItemGroup>
+
+    <WriteLinesToFile File="$(_EmccDefaultsRspPath)"
+                      Lines="@(_EmccFlags)"
+                      WriteOnlyWhenDifferent="true"
+                      Overwrite="true" />
+
+    <!-- Generate Emcc.props -->
+
+    <RunWithEmSdkEnv Command="$(EmccCmd) --version"
+          ConsoleToMsBuild="true"
+          EmSdkPath="$(EMSDK_PATH)"
+          IgnoreStandardErrorWarningFormat="true">
+      <Output TaskParameter="ConsoleOutput" ItemName="_VersionLines" />
+    </RunWithEmSdkEnv>
+
+    <!-- we want to get the first line from the output, which has the version.
+         Rest of the lines are the license -->
+    <ItemGroup>
+      <_ReversedVersionLines Include="@(_VersionLines->Reverse())" />
+    </ItemGroup>
+    <PropertyGroup>
+      <_EmccVersionRaw>%(_ReversedVersionLines.Identity)</_EmccVersionRaw>
+
+      <_EmccVersionRegexPattern>^ *emcc \([^\)]+\) *([^ \(]+) *\(([^\)]+)\)$</_EmccVersionRegexPattern>
+      <_EmccVersion>$([System.Text.RegularExpressions.Regex]::Match($(_EmccVersionRaw), $(_EmccVersionRegexPattern)).Groups[1].Value)</_EmccVersion>
+      <_EmccVersionHash>$([System.Text.RegularExpressions.Regex]::Match($(_EmccVersionRaw), $(_EmccVersionRegexPattern)).Groups[2].Value)</_EmccVersionHash>
+
+      <_EmccPropsContent>
+<![CDATA[
+<Project>
+  <PropertyGroup>
+    <RuntimeEmccVersionRaw>$(_EmccVersionRaw)</RuntimeEmccVersionRaw>
+    <RuntimeEmccVersion>$(_EmccVersion)</RuntimeEmccVersion>
+    <RuntimeEmccVersionHash>$(_EmccVersionHash)</RuntimeEmccVersionHash>
+  </PropertyGroup>
+</Project>
+]]>
+      </_EmccPropsContent>
+    </PropertyGroup>
+
+    <Error Text="Failed to parse emcc version, and hash from the full version string: '$(_EmccVersionRaw)'"
+           Condition="'$(_EmccVersion)' == '' or '$(_EmccVersionHash)' == ''" />
+
+    <WriteLinesToFile File="$(NativeBinDir)src\Emcc.props"
+                      Lines="$(_EmccPropsContent)"
+                      Overwrite="true"
+                      WriteOnlyWhenDifferent="true" />
+  </Target>
+
   <!-- This is a documented target that is invoked by developers in their innerloop work. -->
   <Target Name="BuildWasmRuntimes"
           Condition="'$(OS)' != 'Windows_NT'"
           AfterTargets="Build"
-          DependsOnTargets="BuildPInvokeTable;BundleTimeZones">
+          DependsOnTargets="GenerateEmccPropsAndRspFiles;BuildPInvokeTable;BundleTimeZones">
 
     <Exec Command="make -C $(MonoProjectRoot)wasm all SHELL=/bin/bash BINDIR=$(ArtifactsBinDir) MONO_BIN_DIR=$(MonoArtifactsPath) OBJDIR=$(ArtifactsObjDir) NATIVE_BIN_DIR=$(NativeBinDir) CONFIG=$(Configuration) PINVOKE_TABLE=$(WasmPInvokeTablePath) ICU_LIBDIR=$(ICULibDir) ENABLE_ES6=$(WasmEnableES6) ENABLE_METADATA_UPDATE=$(WasmEnableMetadataUpdate)"
```

Comment:
I think `ENABLE_ES6=$(WasmEnableES6) ENABLE_METADATA_UPDATE=$(WasmEnableMetadataUpdate)` should be removed, right?

Reply:
I mean it so, that we should remove them in this PR context. If I understand it correctly, we are not handling these in the Makefile anymore, so we don't need to pass them here. Hopefully I didn't overlook something.